src/dune_pkg/substs.mli

```diff
@@ -0,0 +1,25 @@
+open Stdune
+
+module Var : sig
```

Comment:
It seems like we're defining lots of different types of variables in different places. I'm currently looking at expanding variables in build/install commands and the set of available variables in substs is the same. I was planning to represent each variable as a pform and then use the existing pform expander to resolve variables to their values. Would that work for substitutions as well?

Reply:
Yes, I assume it would work as well, the only thing required there is a function that translates from an opam "full variable" to a string with the semantics of OPAM (like rendering bools and string lists).